plugins/database/mongodb/mongodb.go

```diff
@@ -124,7 +127,21 @@ func (m *MongoDB) CreateUser(statements dbplugin.Statements, usernameConfig dbpl
 	}
 
 	err = session.DB(mongoCS.DB).Run(createUserCmd, nil)
-	if err != nil {
+	switch err {
+	case nil:
+	case io.EOF || strings.Contains(err.Error(), "EOF"):
```

Comment:
Did you mean to compare `err` with `io.EOF`? I wonder if this will compile since `io.EOF` does not return a bool.

Reply:
`io.EOF` is the following:

`var EOF = errors.New("EOF")`

As such I am indeed trying to compare it directly, but in case it's simply another actual error type that contains EOF in its string, I'm also doing the string check.

Reply:
Since its a error object and not a bool, I suspect if this will compile.

Reply:
Should it be a comma instead of a `||`?

Reply:
So you're totally right about that, which is really amusing because the original reporter swore that he'd tried this branch and it didn't fix his problem. I'm suspicious.